stable/datadog/templates/daemonset.yaml

```diff
@@ -50,14 +54,24 @@ spec:
         {{- if .Values.datadog.apmEnabled }}
           {{- include "container-trace-agent" . | nindent 8 }}
         {{- end }}
-        {{- if .Values.datadog.processAgentEnabled }}
+        {{- if or (kindIs "invalid" .Values.datadog.processAgentEnabled) .Values.datadog.processAgentEnabled }}
```

Comment:
This is because process-agent has 3 states: if the `enabled` flag is missing, process-agent would run, but only doing container checks. If `enabled: true`, process-agent does container and process checks. If `enabled: false`, then process-agent is not running. In helm there's no way other than this ☝️ to check if a config value is `nil`.

Reply:
In that case I would suggest to use two booleans to configure the `processAgent` as you describe; it is to be consistent with other components chart configuration. 

(for example the `cluster-agent` with `metricsProvider` and `clusterChecks`) . https://github.com/helm/charts/blob/master/stable/datadog/values.yaml#L283

Reply:
I guess `processAgentEnabled` wasn't the best name we could come up with so 😅. We should clarify the doc on this IMO. I think that we have two options:
- Keep those changes so we propagate this kind of weird 3-state parameter to the new multi container approach
- Document the fact that in a multi-container environment it enables or disables the entire process (container) and add `daemonsets.containers.processAgent.X` options with clearer names and use them as proper booleans in the chart.

Reply:
Thanks @clamoriniere @xlucas for the discussion, I tend to keep the logic here, because it is consistent with the usage of env variable `DD_PROCESS_AGENT_ENABLED`(the env var can be absent, true, or false). I should definitely put documentation in README to clarify the situation, but I hope that would be enough.

Reply:
I updated the doc here addressing the usage: https://github.com/helm/charts/pull/14674/files#diff-d1853bc077f8540ce703988dee21395dR255